src/esp/geo/VoxelGrid.cpp

```diff
@@ -0,0 +1,340 @@
+
+#include <assert.h>
+#include <limits.h>
+#include <cmath>
+
+#include <Corrade/Utility/Algorithms.h>
+#include <Magnum/GL/DefaultFramebuffer.h>
+#include <Magnum/MeshTools/Interleave.h>
+#include <Magnum/MeshTools/Reference.h>
+#include <Magnum/Primitives/Cube.h>
+
+#include "VoxelGrid.h"
+#include "esp/assets/ResourceManager.h"
+
+namespace Mn = Magnum;
+namespace Cr = Corrade;
+
+namespace esp {
+namespace geo {
+
+#ifdef ESP_BUILD_WITH_VHACD
+VoxelGrid::VoxelGrid(const std::unique_ptr<assets::MeshData>& meshData,
```

Comment:
The  `meshData` can't be null, right? Then maybe it would make sense to accept just 

```suggestion
VoxelGrid::VoxelGrid(const assets::MeshData& meshData,
```

instead.